seshdash/views.py

```diff
@@ -724,16 +724,31 @@ def historical_data(request):
 #function for Graph Generations   
 @login_required
 def graphs(request):
+
+    # if ajax request
     if request.method == 'POST':
+
+        # variables declaration
         results = {}
+        time_delta_dict = {}
+        time_bucket_dict = {}
         data_values =[]
         date_time = []
+
+        # Getting values from Post request
         time = request.POST.get('time','')
         choices = request.POST.getlist('choice[]')
         active_id = request.POST.get('active_site_id','')
         active_site = Sesh_Site.objects.filter(id=active_id)
-        active_site_name = active_site [0]
+       
+        # Checking for a valid site_id
+        if active_site != []:
+            active_site_name = active_site [0]
+        else:
+            return HttpResponseBadRequest()
         current_site = active_site_name.site_name
+
+        # processing post request values to be used in the influx queries
         for choice in choices:
             data_values = []
             time_delta_dict = {'24h':{'hours':24},'7d':{'days':7},'30d':{'days':30}}
```

Comment:
can you also move time delta dict to beginning of the function.
Is there need for it to be declared in every iteration of the `for` loop? @Frankkwizera